src/transforms/onehotencoding.jl

```diff
@@ -0,0 +1,86 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    OneHotEncoding(col)
+    
+Transforms categorical column `col` into one-hot columns of levels
+returned by the `levels` function of CategoricalArrays.jl.
+
+# Examples
+
+```julia
+OneHotEncoding(1)
+OneHotEncoding(:a)
+OneHotEncoding("a")
+```
+"""
+struct OneHotEncoding{S<:ColSelector} <: Stateless
+  col::S
+end
+
+isrevertible(::Type{<:OneHotEncoding}) = true
+
+_colname(col::Integer, names) = names[col]
+_colname(col::AbstractString, names) =
+  _colname(Symbol(col), names)
+
+function _colname(col::Symbol, names)
+  @assert col ∈ names "Invalid column."
+  return col
+end
+
+_name(nm, names) = 
+  nm ∈ names ? _name(Symbol("$(nm)_"), names) : nm
+
+function apply(transform::OneHotEncoding, table)
+  cols = Tables.columns(table)
+  names = collect(Tables.columnnames(cols))
+  columns = AbstractVector[Tables.getcolumn(cols, nm) for nm in names]
+  
+  name = _colname(transform.col, names)
+  ind = findfirst(==(name), names)
+  x = columns[ind]
+
+  if !isa(x, CategoricalArray)
+    throw(ArgumentError("The $name column must be cetegorical."))
+  end
+
+  xl = levels(x)
+  onehot = map(xl) do l
+    nm = Symbol("$(name)_$l")
+    nm = _name(nm, names)
+    nm, x .== l
+  end
+
+  newcols, newnms = last.(onehot), first.(onehot)
```

Comment:
Can you please swap the order here? Why you create a pair with last/first instead of first/last which is more natural? Also, we always think of these pairs as names => cols, not cols => names. Let's try to follow this pattern.